src/common/common.js

```diff
@@ -57,17 +57,32 @@ const common = {
     const result = new BigNumber(satoshi).div('1e8');
     return result;
   },
-  rskCoinToWeiHex(amount) {
-    const result = `0x${this.rskCoinToWei(amount).decimalPlaces(0).toString(16)}`;
-    return result;
+  rskCoinToWeiHex(amount, precision) {
+    const wei = this.rskCoinToWei(amount, precision);
+    return Rsk3.utils.numberToHex(wei);
   },
-  rskCoinToWei(amount) {
-    const result = new BigNumber(amount).times('1e18');
-    return result;
+  /**
+   * Transform coin value to wei unit
+   * Example: RIF, precision = 18, 0.001 RIF = 1e15 (wei) RIF
+   * Example: SOL, precision = 2, 1 SOL = 100 (wei) SOL
+   * @param {*} amount
+   * @param {*} precision
+   */
+  rskCoinToWei(amount, precision = 18) {
+    const precisionInteger = _.floor(Number(precision));
```

Comment:
instead of floor, which masks input errors, this should reject non integer values explicitly, throwing an error. I suggest that you use this to test: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

additionally negative values for precision are not legal according to the EIP (unsigned integer in Solidity), so you need to also add a check that rejects values less than zero.